---
layout: page
title: "Annotation Training"
permalink: /annotation/
tags: [ gene annotation ]
description: Gene Annotation page
---

<table class="table-contact">
<tr>
<td><img width="100%" class="img-responsive" src="{{ site.url }}/images/pga/logo_gnpannot-v2.png" alt="" />
</td>
<td>
This course introduces annotation, such as structural gene annotation with EGNEP or MAKER workflows, in the Plant Genomme Assembly and Annotation course, both in a theoretical and practical way, with hands-on practice sessions. It will help you to configure and run your own pipelines using these tools through command-line interface.
</td>
</tr>
</table>

<small>MAKER2 : An annotation pipeline and genome-database management tool for second-generation genome projects. Holt, Carson; Yandell, Mark. BMC Bioinformatics, Vol. 12, No. 1, 491, 22.12.2011. https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-491.

<small>Genome Annotation in Plants and Fungi: EuGène as a Model Platform. Sylvain Foissac et al. Current Bioinformatics, 2008. DOI: 10.2174/157489308784340702

<small>Artemis: an integrated platform for visualization and analysis of high-throughput sequence-based experimental data.
Carver T, Harris SR, Berriman M, Parkhill J and McQuillan JA. Bioinformatics (Oxford, England) 2012;28;4;464-9
[PMID:22199388](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-491)

[MAKER website](http://www.yandell-lab.org/software/maker.html)

[EuGène website](http://eugene.toulouse.inra.fr/)

[Artemis](https://www.sanger.ac.uk/science/tools/artemis)

### Prerequisites
Prior knowledge on workflow execution not necessary 
Basic knowledge of Linux (Linux for dummies required)

<div id="colonne1">
<h3>Program</h3>
<ul>
<li>Why using a combiner / integrator to annotate gene ?</li>
<li>How to annotate a eukaryotic genome ?</li>
<li>How to configure / to customize gene annotation workflow ? </li>
<li>How to execute them? </li>
<li>Use case</li>
</ul>
</div>

<div id="colonne2">
<h3>Learning objectives</h3>
<ul>
<li>Have revised the structure of an eukaryotic gene.</li>
<li>Learned what is annotation of eukaryotic genes.</li>
<li>Understood the major steps of a combiner like EuGene.</li> 
<li>Explaining the major steps of MAKER and EuGène,<br>in which way they differ from each other.</li>  
<li>Annotating your genome with MAKER and EuGène.</li>
<li>Know how to Launch a virtual machine on IFB cloud.</li>
<li>Know how to configure and run MAKER VM / Eugene VM on the Arabidopsis genome.</li>
<li>Know how to Assess the results with Busco.</li>
</ul>
</div>

<div id="colonne3">
<h3>Training material</h3>
<ul>
<li>Practice Annotation MAKER : <a target="_blank" href="{{ site.url }}/annotation/MAKER/StructuralAnnotation_mtp">see</a></li>
<li>Practice Annotation Eugene : <a target="_blank" href="{{ site.url }}annotation/Eugene/exercice_eugene_appliance">see</a></li><li>Practice Manual Curation : <a target="_blank" href="{{ site.url }}/annotation/artemis_manual_curation">see</a></li> 
</ul>
</div>

<div id="nextInline" class="clearfix">
<h3>Instructors</h3>
<ul>
<li>Jacques Dainat (JD) - jacques.dainat@nbis.se</li>
<li>Lucile Soler (LS) - lucile.soler@nbis.se</li>
<li>Stéphanie Bocs (SSB) - stephanie.sidibe-bocs@cirad.fr</li>
<li>Laurent Bouri (LB) - Laurent.BOURI@france-bioinformatique.fr</li>
<li>Jonathan Lorenzo (JL) - jonathan.lorenzo@france-bioinformatique.fr</li></ul>
</div>

### Trainings

| **Date** | **Location** | **Topics** | **Participants** | **Instructors** | **Links** | **Units** |
| 2018 | Montpellier, France | PGAA  | 22 | JD, LS, SB, LB, JL | [Elixir GAA wiki](https://biosphere.france-bioinformatique.fr/wikia2/index.php/Main_Page) [PGAA course announcement](https://www.france-bioinformatique.fr/fr/evenements/plant-genome-assembly-and-annotation) [Elixir e-learning platform](https://elixir.mf.uni-lj.si/course/view.php?id=33) | IRD, Cirad, Montpellier SupAgro, INRA, IPS2, Neiker, RWTH Aachen University |




