<!doctype html>
<!--[if lt IE 7]><html class="no-js ie6 oldie" lang="ja"><![endif]-->
<!--[if IE 7]>   <html class="no-js ie7 oldie" lang="ja"><![endif]-->
<!--[if IE 8]>   <html class="no-js ie8 oldie" lang="ja"><![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="ja"><!--<![endif]-->
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

    <title>RNASeq Practice</title>
    <meta name="description" content="Bioinformatics trainings, Linux, HPC, NGS, NGS workflow manager, TOGGLe, galaxy">
    <meta name="description" content="RNASeq Practice page">
    
    <meta name="author" content=" Sébastien RAVEL -- sebastien.ravel@cirad.fr">
    
    <meta name="author" content=" Christine Tranchant-Dubreuil -- christine.tranchant@ird.fr">
    

    <meta name="viewport" content="width=device-width,initial-scale=1">

    <link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/bootstrap.min.css" >
    <link rel="stylesheet" type="text/css" media="screen" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/monokai.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://toggle.ird.fr:4000/css/stylesheet.css">
    <link rel="icon" href="http://toggle.ird.fr:4000/images/southgreenlong.png">
</head>
 <body>

      <!-- Static navbar -->
      <nav class="navbar navbar-default navbar-inverse navbar-static-top">
        <div class="container">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
              <span class="sr-only"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand visible-xs" href="#"></a>
          </div>
          <div id="navbar" class="navbar-collapse collapse">
			<ul id="menu">
				<li><a href="http://toggle.ird.fr:4000/">Home</a></li>
				<li><a href="#">Linux</a><span class="caret" ></span>
					<ul id="child">
						<li><a href="http://toggle.ird.fr:4000/linux/">For Dummies</a></li>
						<li><a href="http://toggle.ird.fr:4000/linuxJedi/">For Jedi</a></li>
					</ul>
				</li>
				<li><a href="http://toggle.ird.fr:4000/perl/">Perl</a></li>
    <!--<li><a href="http://toggle.ird.fr:4000/R/">R</a></li>-->
				<li><a href="http://toggle.ird.fr:4000/HPC/">HPC</a></li>
				</li><li><a href="#">Workflow</a><span class="caret" ></span>
					<ul>
						<li><a href="http://toggle.ird.fr:4000/toggle/">TOGGLe</a></li>
						<li><a href="http://toggle.ird.fr:4000/galaxy/">Galaxy</a></li>
						<li><a href="http://toggle.ird.fr:4000/gigwa/">GIGWA</a></li>
						<li><a href="http://toggle.ird.fr:4000/galaxyToggle/">Workflow Manager</a></li>

					</ul>
				</li>
                               <li><a href="http://toggle.ird.fr:4000/ngs/">NGS</a><span class="caret" ></span>
                                        <ul>
                                              <!--   <li><a href="http://toggle.ird.fr:4000/todo/">GBS</a></li>-->
                                                <li><a href="http://toggle.ird.fr:4000/rnaseq/">RNASeq</a></li>
                                                <li><a href="http://toggle.ird.fr:4000/metabarcoding/">Meta- barcoding</a></li>
						<li><a href="http://toggle.ird.fr:4000/gbsGuadeloupe/">GBS</a></li>
                                        </ul>
                                </li>
	<!--			<li><a href="#">Schools</a><span class="caret" ></span>
                                        <ul>
                                                <li><a href="http://toggle.ird.fr:4000/todo/">Musa genome</a></li>
                                                <li><a href="http://toggle.ird.fr:4000/todo/">France</a></li>
                                                <li><a href="http://toggle.ird.fr:4000/todo/">International</a></li>
                                        </ul>
                                </li>-->
				<li><a href="http://toggle.ird.fr:4000/contact/">About Us</a></li>
				<li><a id="forkme_banner" href="https://github.com/SouthGreenPlatform/trainings/">GitHub</a></li>
			</ul>
          </div>
        </div>
      </nav>
      <!--/.nav-collapse -->
    <!-- Header -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <div class="header"><img class="img-responsive" width="25%" src="http://toggle.ird.fr:4000/images/southgreenlong.png" alt="South Green Logo" /></div>
          <h2 id="project_tagline">South Green Trainings pages</h2>
        </header>
    </div>

<body>
<div id="main_content_wrap" class="outer">
	 <section id="main_content" class="inner">
<table>
  <thead>
    <tr>
      <th style="text-align: left">Description</th>
      <th style="text-align: left">Hands On Lab Exercises for RNASeq</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Related-course materials</td>
      <td style="text-align: left"><a href="https://southgreenplatform.github.io/tutorials//bioanalysis/rnaSeq/">Transcriptomique</a></td>
    </tr>
    <tr>
      <td style="text-align: left">Authors</td>
      <td style="text-align: left">Julie Orjuela (julie.orjuela@irf.fr), Gautier Sarah (gautier.sarah@cirad.fr), Catherine Bréton (c.breton@cgiar.org), Aurore Comte (aurore.compte@ird.fr),  Alexis Dereeper (alexis.dereeper@ird.fr), Sebastien Ravel (sebastien.ravel@cirad.fr), Sebastien Cunnac (sebastien.cunnac@ird.fr)</td>
    </tr>
    <tr>
      <td style="text-align: left">Creation Date</td>
      <td style="text-align: left">15/03/2018</td>
    </tr>
    <tr>
      <td style="text-align: left">Last Modified Date</td>
      <td style="text-align: left">05/04/2018</td>
    </tr>
  </tbody>
</table>

<hr />

<h3 id="summary">Summary</h3>

<!-- TOC depthFrom:2 depthTo:2 withLinks:1 updateOnSave:1 orderedList:0 -->
<ul>
  <li><a href="#practice-1">Practice 1: Psedo-mapping against transcriptome reference + counting with Kallisto</a></li>
  <li><a href="#practice-2">Practice 2: Mapping against annotated genome reference with Hisat2 + counting with Stringtie</a></li>
  <li><a href="#practice-3">Practice 3: Differential expression analysis using EdgeR and DESeq2</a></li>
  <li><a href="#practice-4">Practice 4: Visualization of mapped reads against genes using IGV</a></li>
  <li><a href="#practice-5">Practice 5: Explore multiple expression projects/experiments using DiffExDB ? ? ? ? ? </a></li>
  <li><a href="#practice-6">Practice 6: Heatmap and Hierarchical Clustering</a></li>
  <li><a href="#practice-7">Practice 7: Co-expression network analysis</a></li>
  <li><a href="#links">Links</a></li>
  <li><a href="#license">License</a></li>
</ul>

<hr />

<p><a name="practice-1"></a></p>
<h3 id="practice-1--mapping-against-transcriptome-reference--counting-with-kallisto">Practice 1 : Mapping against transcriptome reference + counting with Kallisto</h3>
<table class="table-contact">
<tr>
<td>Practice1 will be performed in the Galaxy environment.</td>
<td><img width="60%" src="http://toggle.ird.fr:4000/images/trainings-galaxy.png" alt="" />
</td>
</tr>
</table>
<p>We will perform a transcriptome-based mapping and estimates of transcript levels using Kallisto, and a differential analysis using EdgeR.</p>
<ul>
  <li>Connect to <a href="http://bioinfo-inter.ird.fr:8080/">Galaxy IRD</a></li>
  <li>Create a new history and import RNASeq samples datasets (paired-end fastq files) from Data library
<code class="highlighter-rouge">formation Galaxy 2019 / RNAseq</code></li>
  <li>Upload the Chr1 of rice transcriptome (cDNA) to be used as reference  - <code class="highlighter-rouge">http://rice.plantbiology.msu.edu/pub/data/Eukaryotic_Projects/o_sativa/annotation_dbs/pseudomolecules/version_7.0/chr01.dir/Chr1.cdna</code></li>
  <li>Run the kallisto program by providing Chr1 as transcriptome reference and specifying correctly pairs of input fastq- <code class="highlighter-rouge">kallisto quant</code></li>
  <li>Convert kallisto outputs (collection of count files) into one single file that can be used as input for EdgeR - <code class="highlighter-rouge">Kallisto2EdgeR</code></li>
</ul>

<hr />

<p><a name="practice-2"></a></p>
<h3 id="practice-2--mapping-against-annotated-genome-reference-with-hisat2--counting-with-stringtie">Practice 2 : Mapping against annotated genome reference with Hisat2 + counting with Stringtie</h3>
<p><img width="20%" src="http://toggle.ird.fr:4000/images/toggleLogo2.png" alt="" /></p>

<ul>
  <li>Running Hisat2 and Stringtie with TOGGLe</li>
</ul>

<p>Connect to account in IRD i-Trop cluster:</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">ssh formationX@bioinfo-master.ird.fr</code></pre></figure>

<p>Input data are accessible from :</p>

<ul>
  <li>Input data : /data2/formation/tp-toggle/RNASeqData/</li>
  <li>Reference : /data2/formation/tp-toggle/RNASeqData/referenceFiles/chr1.fasta</li>
  <li>Config file: <a href="https://raw.githubusercontent.com/SouthGreenPlatform/TOGGLE/master/exampleConfigs/RNASeqHisat2Stringtie.config.txt">RNASeqReadCount.config.txt</a></li>
</ul>

<p>Before to start …</p>

<ul>
  <li>Create a toggleTP directory in your HOME</li>
  <li>Make à copy for reference and input data into toggleTP directory (cp).</li>
  <li>Add the configuration file used by TOGGLe and change SGE key as below</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">$sge</span>
<span class="nt">-q</span> bioinfo.q
<span class="nt">-b</span> Y
<span class="nt">-cwd</span></code></pre></figure>

<p>SOLUTION:</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">mkdir ~/toggleTP
<span class="nb">cd</span> ~/toggleTP
cp /data2/formation/tp-toggle/RNASeqData/ ./ <span class="nt">-r</span>
wget https://raw.githubusercontent.com/SouthGreenPlatform/TOGGLE/master/exampleConfigs/RNASeqHisat2Stringtie.config.txt
vim RNASeqHisat2Stringtie.config.txt</code></pre></figure>

<p>Your data are now in ~/toogleTP.  Now, create a <code class="highlighter-rouge">runTOGGLeRNASEQ.sh</code> bash script to launch TOGGLe :</p>

<p>SOLUTION:</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c">#!/bin/bash</span>
<span class="c">#$ -N TOGGLeRNAseq</span>
<span class="c">#$ -b yes</span>
<span class="c">#$ -q bioinfo.q</span>
<span class="c">#$ cwd</span>
<span class="c">#$ -V</span>

<span class="nv">dir</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/fastq"</span>
<span class="nv">out</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/outTOGGLe"</span>
<span class="nv">config</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/RNASeqHisat2Stringtie.config.txt"</span>
<span class="nv">ref</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/referenceFiles/chr1.fasta"</span>
<span class="nv">gff</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/Chr1.gff3"</span> <span class="c">#verifier si gff3 dispo</span>

<span class="c">## Software-specific settings exported to user environment</span>
module load bioinfo/TOGGLE-dev/0.3.7

<span class="c">#running tooglegenerator </span>
toggleGenerator.pl <span class="nt">-d</span> <span class="nv">$dir</span> <span class="nt">-c</span> <span class="nv">$config</span> <span class="nt">-o</span> <span class="nv">$out</span> <span class="nt">-r</span> <span class="nv">$ref</span> <span class="nt">-g</span> <span class="nv">$gff</span> <span class="nt">--report</span><span class="p">;</span>

<span class="nb">echo</span> <span class="s2">"FIN"</span></code></pre></figure>

<p>This is the software configutation to create a TOGGLe pipeline with Hisat2 and Stringtie. You can check parametters of every step here. You can open it with Komodo Edit.
<code class="highlighter-rouge">vim ~/toggleTP/RNASeqData/RNASeqHisat2Stringtie.config.txt</code></p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">$order</span>
<span class="nv">1</span><span class="o">=</span>fastqc
<span class="nv">2</span><span class="o">=</span>hisat2
<span class="nv">3</span><span class="o">=</span>samtoolsView
<span class="nv">4</span><span class="o">=</span>samtoolsSort
<span class="nv">5</span><span class="o">=</span>stringtie 1
<span class="nv">1000</span><span class="o">=</span>stringtie 2

<span class="nv">$samtoolsview</span>
<span class="nt">-b</span>
<span class="nt">-h</span>

<span class="nv">$samtoolssort</span>

<span class="nv">$cleaner</span>
3

<span class="c">#PUT YOUR OWN SGE CONFIGURATION HERE</span>
<span class="nv">$sge</span>
<span class="nt">-q</span> bioinfo.q
<span class="nt">-b</span> Y

<span class="nv">$stringtie</span> 1

<span class="nv">$stringtie</span> 2
<span class="nt">--merge</span>

<span class="nv">$hisat2</span>
<span class="nt">--dta</span>

<span class="nv">$scp</span>
/scratch/

<span class="nv">$env</span>
module load bioinfo/TOGGLE-dev/0.3.7</code></pre></figure>

<p>Command line generates by <code class="highlighter-rouge">runTOGGLeRNASEQ.sh</code> is :</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">/usr/local/TOGGLE-dev-0.3.7/toggleGenerator.pl <span class="nt">-d</span> <span class="s2">"~/toggleTP/RNASeqData/fastq -c ~/toggleTP/RNASeqData/RNASeqHisat2Stringtie.config.txt -o ~/toggleTP/RNASeqData/outTOGGLe -r ~/toggleTP/RNASeqData/referenceFiles/chr1.fasta -g ~/toggleTP/RNASeqData/Chr1.gff3 --report --nocheck</span></code></pre></figure>

<p>Launch runTOGGLeRNASEQ.sh in qsub mode</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">qsub <span class="nt">-q</span> bioinfo.q <span class="nt">-N</span> TOGGLeRNASEQ <span class="nt">-b</span> yes <span class="nt">-cwd</span> <span class="s1">'module load bioinfo/TOGGLE-dev/0.3.7; ./runTOGGLeRNASEQ.sh '</span></code></pre></figure>

<ul>
  <li>
    <p>Explore output TOGGLe and check if everything was ok</p>
  </li>
  <li>
    <p>Run again stringtie using options -B and -e</p>
  </li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">OUTPUT</span><span class="o">=</span><span class="s2">"~/toggleTP/RNASeqData/outTOGGLe/"</span>
mkdir <span class="nv">$OUTPUT</span>/stringtieEB
<span class="nb">cd</span> <span class="nv">$OUTPUT</span>/stringtieEB 
ln <span class="nt">-s</span> <span class="nv">$OUTPUT</span>/finalResults/intermediateResults.STRINGTIEMERGE.gtf <span class="nb">.</span>
ln <span class="nt">-s</span> <span class="nv">$OUTPUT</span>/output/<span class="k">*</span>/4_samToolsSort/<span class="k">*</span>SAMTOOLSSORT.bam .</code></pre></figure>

<ul>
  <li>Remember good pratices to work at IRD Cluster. You have to copy data into a path /scratch in a node.</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">qrsh <span class="nt">-q</span> formation.q
scp nas:<span class="nv">$OUTPUT</span>/stringtieEB /scratch/formation1 
<span class="nb">cd</span> /scratch/formation1 </code></pre></figure>

<ul>
  <li>… before merging gtf files obtained by stringtie, we have to recovery annotations in order to see gene name in gtf files. Stringtie annotate transcrips using gene id ‘MSTRG.1’ nomenclature . See https://github.com/gpertea/stringtie/issues/179</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">python3 ~/scripts/gpertea-scripts/mstrg_prep.py intermediateResults.STRINGTIEMERGE.gtf <span class="o">&gt;</span> intermediateResults.STRINGTIEMERGE_prep.gtf </code></pre></figure>

<p>Compare output before and after run <code class="highlighter-rouge">mstrg_prep.py</code></p>

<ul>
  <li>… Now we launch stringtie:</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for </span>i <span class="k">in</span> <span class="se">\*</span>bam <span class="p">;</span> <span class="k">do </span><span class="nb">echo</span> <span class="s2">"mkdir </span><span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.bam/</span><span class="k">}</span><span class="s2">; qsub -q bioinfo.q -N stringtie2 -cwd -V -b yes 'module load bioinfo/stringtie/1.3.4; stringtie"</span> <span class="nv">$PWD</span><span class="s2">"/"</span><span class="nv">$i</span> <span class="s2">"-G </span><span class="nv">$PWD</span><span class="s2">"</span>/<span class="s2">"intermediateResults.STRINGTIEMERGE_prep.gtf -e -B -o </span><span class="nv">$PWD</span><span class="s2">/</span><span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.bam/</span><span class="k">}</span><span class="s2">/</span><span class="k">${</span><span class="nv">i</span><span class="p">/bam/count</span><span class="k">}</span><span class="s2">'"</span><span class="p">;</span> <span class="k">done</span></code></pre></figure>

<ul>
  <li>Convert stringtie output in counts using <code class="highlighter-rouge">prepDE.py</code>. Dont forget. You are in /scratch <code class="highlighter-rouge">/scratch/formation1 </code></li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">mkdir counts
<span class="nb">cd </span>counts/
ln <span class="nt">-s</span> <span class="nv">$OUTPUT</span>/stringtieEB/<span class="k">*</span>/<span class="k">*</span>.count <span class="nb">.</span>
<span class="k">for </span>i <span class="k">in</span> <span class="se">\*</span>.count<span class="p">;</span> <span class="k">do </span><span class="nb">echo</span> <span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.count/</span><span class="k">}</span> <span class="nv">$PWD</span>/<span class="nv">$i</span><span class="p">;</span> <span class="k">done</span> <span class="o">&gt;</span> listGTF.txt
python2 /data/projects/TALseq/stringtie-scripts/prepDE.py <span class="nt">-i</span> listGTF.txt</code></pre></figure>

<ul>
  <li>Don’t forget scp *.counts files to you $OUTPUT</li>
</ul>

<hr />

<p><a name="practice-3"></a></p>
<h3 id="practice-3--differential-expression-analysis-using-edger-and-deseq2">Practice 3 : Differential expression analysis using EdgeR and DESeq2</h3>
<td>Practice3 will be performed in PIVOT via R Studio.</td>
<ul>
  <li>Run the EdgeR program for differential analysis - <code class="highlighter-rouge">edger</code></li>
  <li>Verify relevance of normalized expression values provided by EdgeR</li>
  <li>Observe MDS plot of experimental conditions. Observe Smear plot.</li>
  <li>
    <p>Using <code class="highlighter-rouge">sort</code> and  its <code class="highlighter-rouge">general numeric sort</code> parameter, combined with <code class="highlighter-rouge">filter</code> tool, determine how many genes are found to be differentally expressed using a minimum pvalue &lt;= 0.05? Using a minimum FDR-adjusted pvalue &lt;= 0.05?</p>
  </li>
  <li>
    <p>Open PIVOT and run DESeq2 after having cutting and importing columns for each conditions.</p>
  </li>
  <li>Go to file Tab.</li>
  <li>Take the count file <code class="highlighter-rouge">gene_count_matrix.csv</code> generated previously.</li>
  <li>Import this file into  Data input  and then Input file type.</li>
  <li>Add Use Tab separator to Skip Rows.</li>
  <li>
    <p>Check if yours data are imported in the rigth window.</p>
  </li>
  <li>Go to Design tab.</li>
  <li>Upload the designed table. <code class="highlighter-rouge">info.txt</code></li>
  <li>
    <p>Verify that the header of the info file corresponds to the count file.</p>
  </li>
  <li>Go To sample</li>
  <li>
    <p>Select your sample and condition</p>
  </li>
  <li>Go to Basic statistic</li>
  <li>If you want to keep the normalized, log10 count table, upload it.</li>
  <li>
    <p>Verify the distribution of each condition in the standard deviation graph, the dispersion graph.</p>
  </li>
  <li>Go to Differential expression</li>
  <li>Launch DESeq2, experimental design, and</li>
  <li>How many…..</li>
  <li>How many genes are differentially expressed at early stage of infection (2 day) in complete genome?</li>
  <li>How many genes are induced by the presence of nematode with logFC &gt; 2? How many genes are repressed?</li>
  <li>How many genes are induced at both early and late stages of infection?</li>
  <li>How many genes are repressed gradually along infection? Have a look at heatmap representation for these genes?</li>
</ul>

<hr />

<ul>
  <li>Connect to <a href="http://bioinfo-inter.ird.fr:8080">IRD Galaxy</a> and run DESeq2 after having cutting and importing columns for each conditions. Determine how many genes are found to be differentally expressed using a minimum pvalue &lt;= 0.05? Using a minimum FDR-adjusted pvalue &lt;= 0.05?</li>
</ul>

<hr />

<ul>
  <li>Compare lists of DE genes with the two approches using <a href="http://bioinfogp.cnb.csic.es/tools/venny/">Venny</a>. Look at the expression values for a gene found DE with EdgeR and not with DESeq2, and vice-versa.</li>
</ul>

<hr />

<p><a name="practice-4"></a></p>
<h3 id="practice-4--visualization-of-mapped-reads-against-genes-using-igv">Practice 4 : Visualization of mapped reads against genes using IGV</h3>
<p>Practice3 will be performed with Integrated Genome Viewer (IGV).</p>
<ul>
  <li>Load reference genome, GFF annotation file and two BAM files corresponding to 0dpi and 2dpi</li>
  <li>Focus on a gene that has been shown to be differentially expressed and observe the difference of accumation of reads</li>
</ul>

<hr />

<p><a name="practice-5"></a></p>
<h3 id="practice-5--explore-multiple-expression-projectsexperiments-using-web-sites">Practice 5 : Explore multiple expression projects/experiments using web sites</h3>
<p>Practice5 (first part) will be performed using <a href="http://degust.erc.monash.edu/">Degust</a></p>

<ul>
  <li>After having removed the first line, upload your count file into <a href="http://degust.erc.monash.edu/">Degust</a></li>
  <li>Observe the different plots available</li>
  <li>How many genes can be found DE for a minimum pvalue &lt;= 0.05 and abs(logFC) &gt; 2? Observe the plots.</li>
</ul>

<hr />

<p>Practice5 (second part) will be performed using DiffExDB, a database dedicated to centralize expression projects at IRD</p>

<ul>
  <li>Go to the DiffExDB database: <a href="http://bioinfo-web.mpl.ird.fr/cgi-bin2/microarray/public/diffexdb.cgi">DiffExDB</a>. Select the project  <code class="highlighter-rouge">Response to M.graminicola</code>.</li>
  <li>How many genes are differentially expressed at early stage of infection (2 day) in complete genome? Is there any overrepresented Gene Ontology (GO) term? Look at  volcano plot.</li>
  <li>How many genes are induced by the presence of nematode with logFC &gt; 2? How many genes are repressed?</li>
  <li>How many genes are induced at both early and late stages of infection?</li>
  <li>How many genes are repressed gradually along infection? Have a look at heatmap representation for these genes?</li>
</ul>

<hr />

<p><a name="practice-6"></a></p>
<h3 id="practice-6--hierarchical-clustering">Practice 6 : Hierarchical Clustering</h3>
<p>Practice5 will be performed in the Galaxy environment.</p>
<ul>
  <li>Connect to <a href="http://galaxy.southgreen.fr/galaxy/">Galaxy South Green</a></li>
  <li>Run the plotHeatmap program for heatmap and hierarchical clustering - <code class="highlighter-rouge">plotHeatmap</code>. Using EdgeR output and count file, display heatmap and gene clustering dendrogram on genes having a minimum pvalue &lt;= 0.05 and abs(logFC) &gt; 1</li>
</ul>

<hr />

<p><a name="practice-7"></a></p>
<h3 id="practice-7--co-expression-network-analysis">Practice 7 : Co-expression network analysis</h3>
<p>Practice6 will be performed in the Galaxy environment.</p>
<ul>
  <li>Import appropriate datasets from data libraries <code class="highlighter-rouge">test.count</code> and <code class="highlighter-rouge">Traits.csv</code>: expression values for less genes but more conditions. Run the WGCNA program - <code class="highlighter-rouge">wgcna</code></li>
  <li>Download, install and run the network program - <code class="highlighter-rouge">cytoscape</code> to display networks for each gene cluster.</li>
</ul>

<hr />

<h3 id="links">Links</h3>
<p><a name="links"></a></p>

<ul>
  <li>Related courses : <a href="https://southgreenplatform.github.io/trainings/linuxJedi/">Transcriptomics</a></li>
  <li>Degust : <a href="http://degust.erc.monash.edu/">Degust</a></li>
  <li>MeV: <a href="http://mev.tm4.org/">MeV</a></li>
  <li>MicroScope: <a href="http://microscopebioinformatics.org/">MicroScope</a></li>
  <li>Comparison of methods for differential expression: <a href="https://southgreenplatform.github.io/trainings//files/Comparison_of_methods_for_differential_gene_expression_using RNA-seq_data.pdf">Report</a></li>
</ul>

<hr />

<h3 id="license">License</h3>
<p><a name="license"></a></p>

<div>
The resource material is licensed under the Creative Commons Attribution 4.0 International License (<a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">here</a>).
<center><img width="25%" class="img-responsive" src="http://creativecommons.org.nz/wp-content/uploads/2012/05/by-nc-sa1.png" />
</center>
</div>


	</div> <!-- #main -->
</div> <!-- #main-container -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="http://toggle.ird.fr:4000/js/jquery-1.11.3.min.js"></script>
    <script src="http://toggle.ird.fr:4000/js/bootstrap.min.js"></script>
    <script src="http://toggle.ird.fr:4000/js/homepage.js"></script>
    <script src="http://toggle.ird.fr:4000/js/libs/modernizr-2.0.6.min.js"></script>

</script>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <div id="footer">
			<div class="logo-div" class="clearfix">
				<a class="logo" href="http://www.ird.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_ird.png" alt="IRD Logo"  width="" />
				</a>

				<a class="logo" href="http://www.cirad.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo-cirad.png" alt="CIRAD Logo"   width="" />
				</a>

				<a class="logo" href="http://www.inra.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_inra.png" alt="INRA Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.bioversityinternational.org/">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_bioversity.png" alt="Bioversity Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.supagro.fr/">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/logo_supagro.png" alt="SupAgro Logo"   width=""/>
				</a>
				
				<a class="logo" href="http://www.southgreen.fr">
					<img class="img-logo" src="http://toggle.ird.fr:4000/images/southgreenlong.png" alt="South Green Logo"   width=""/>
				</a>

			</div>
		</div>

		<div class="footer_txt">
			<p>The contents of this website are &copy;&nbsp;2019 under the terms of the Licencied under
				<a href="http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html">CeCill-C</a>
				 and <a href="/blob/master/LICENSE">GPLv3</a>.
			</p>
            <p class="webmaster">WebMaster: <a class="webmaster" target="_blank" href="https://d2gg9evh47fn9z.cloudfront.net/800px_COLOURBOX6415001.jpg">Christine</a>
            &
            <a class="webmaster" target="_blank" href="http://moziru.com/images/fluffy-clipart-silly-monster-13.png">Séb</a></p>
		</div>
      </footer>
    </div>

</body>
</html>
