<!doctype html>
<!--[if lt IE 7]><html class="no-js ie6 oldie" lang="ja"><![endif]-->
<!--[if IE 7]>   <html class="no-js ie7 oldie" lang="ja"><![endif]-->
<!--[if IE 8]>   <html class="no-js ie8 oldie" lang="ja"><![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="ja"><!--<![endif]-->
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

    <title>RNASeq Practice</title>
    <meta name="description" content="Welcome to Galaxy South Green">
    <meta name="description" content="RNASeq Practice page">
    
    <meta name="author" content=" Marilyne Summo -- marilyne.summo@cirad.fr">
    

    <meta name="viewport" content="width=device-width,initial-scale=1">

    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/bootstrap.min.css" >
    <link rel="stylesheet" type="text/css" media="screen" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/monokai.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/stylesheet.css">
    <link rel="icon" href="http://localhost:4000/images/southgreenlong.png">
</head>
 <body>

      <!-- Static navbar -->
      <nav class="navbar navbar-default navbar-inverse navbar-static-top">
        <div class="container">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
              <span class="sr-only"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand visible-xs" href="#"></a>
          </div>
          <div id="navbar" class="navbar-collapse collapse">
			<ul id="menu">
        <li><a href="http://localhost:4000/">Home</a></li>
        <li><a href="http://localhost:4000/galaxysouthgreen/">Galaxy South Green</a></li>
				<li><a href="#">Linux</a><span class="caret" ></span>
					<ul id="child">
						<li><a href="http://localhost:4000/linux/">For Dummies</a></li>
						<li><a href="http://localhost:4000/linuxJedi/">For Jedi</a></li>
					</ul>
				</li>
				<li><a href="http://localhost:4000/python/">Python</a></li>
				<li><a href="http://localhost:4000/perl/">Perl</a></li>
    <!--<li><a href="http://localhost:4000/R/">R</a></li>-->
        <li><a href="#">HPC</a><span class="caret" ></span>
					<ul>
				<li><a href="http://localhost:4000/HPC/">HPC</a></li>
				<li><a href="http://localhost:4000/Advanced_HPC/">Advanced HPC</a></li>
        </ul>
				</li><li><a href="#">Workflow</a><span class="caret" ></span>
					<ul>
						<li><a href="http://localhost:4000/toggle/">TOGGLe</a></li>
						<li><a href="http://localhost:4000/galaxy/">Galaxy</a></li>
						<li><a href="http://localhost:4000/gigwa/">GIGWA</a></li>
						<li><a href="http://localhost:4000/galaxyToggle/">Workflow Manager</a></li>
					</ul>
				</li>
                               <li><a href="http://localhost:4000/ngs/">NGS</a><span class="caret" ></span>
                                        <ul>
                                              <!--   <li><a href="http://localhost:4000/todo/">GBS</a></li>-->
                                                <li><a href="http://localhost:4000/rnaseq/">RNASeq</a></li>
                                                <li><a href="http://localhost:4000/metabarcoding/">Meta- barcoding</a></li>
						<li><a href="http://localhost:4000/gbsGuadeloupe/">GBS</a></li>
                                        </ul>
                                </li>
	<!--			<li><a href="#">Schools</a><span class="caret" ></span>
                                        <ul>
                                                <li><a href="http://localhost:4000/todo/">Musa genome</a></li>
                                                <li><a href="http://localhost:4000/todo/">France</a></li>
                                                <li><a href="http://localhost:4000/todo/">International</a></li>
                                        </ul>
                                </li>-->
				<li><a href="http://localhost:4000/contact/">About Us</a></li>
				<li><a id="forkme_banner" href="https://github.com/SouthGreenPlatform/trainings/">GitHub</a></li>
			</ul>
          </div>
        </div>
      </nav>
      <!--/.nav-collapse -->
    <!-- Header -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <div class="header"><img class="img-responsive" width="25%" src="http://localhost:4000/images/southgreenlong.png" alt="South Green Logo" /></div>
          <h2 id="project_tagline">Welcome to <img style="width: 26px; vertical-align: top;" src="http://localhost:4000/images/galaxyIcon_noText.png"> Galaxy</h2>
        </header>
    </div>

<body>
<div id="main_content_wrap" class="outer">
	 <section id="main_content" class="inner">
<table>
  <thead>
    <tr>
      <th style="text-align: left">Description</th>
      <th style="text-align: left">Hands On Lab Exercises for RNASeq</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Related-course materials</td>
      <td style="text-align: left"><a href="https://southgreenplatform.github.io/tutorials//bioanalysis/rnaSeq/">Transcriptomique</a></td>
    </tr>
    <tr>
      <td style="text-align: left">Authors</td>
      <td style="text-align: left">Julie Orjuela (julie.orjuela@irf.fr), Gautier Sarah (gautier.sarah@cirad.fr), Catherine Breton (c.breton@cgiar.org), Aurore Comte (aurore.compte@ird.fr),  Alexis Dereeper (alexis.dereeper@ird.fr), Sebastien Ravel (sebastien.ravel@cirad.fr), Sebastien Cunnac (sebastien.cunnac@ird.fr)</td>
    </tr>
    <tr>
      <td style="text-align: left">Creation Date</td>
      <td style="text-align: left">15/03/2018</td>
    </tr>
    <tr>
      <td style="text-align: left">Last Modified Date</td>
      <td style="text-align: left">17/05/2019</td>
    </tr>
  </tbody>
</table>

<hr />

<h3 id="summary">Summary</h3>

<!-- TOC depthFrom:2 depthTo:2 withLinks:1 updateOnSave:1 orderedList:0 -->
<ul>
  <li><a href="#practice-1">Practice 1: Pseudo-mapping against transcriptome reference + counting with Kallisto</a></li>
  <li><a href="#practice-2">Practice 2: Mapping against annotated genome reference with Hisat2 + counting with Stringtie</a></li>
  <li><a href="#practice-3">Practice 3: Differential expression analysis using EdgeR and DESeq2</a></li>
  <li><a href="#practice-4">Practice 4: Compare list of DE genes with EdgeR and DESeq2</a></li>
  <li><a href="#practice-5">Practice 5: Hierarchical Clustering</a></li>
  <li><a href="#practice-6">Practice 6: Visualization of mapped reads against genes using IGV</a></li>
  <li><a href="#links">Links</a></li>
  <li><a href="#license">License</a></li>
</ul>

<hr />

<p><a name="practice-1"></a></p>
<h3 id="practice-1--mapping-against-transcriptome-reference--counting-with-kallisto">Practice 1 : Mapping against transcriptome reference + counting with Kallisto</h3>
<table class="table-contact">
<tr>
<td>Practice1 will be performed in the Galaxy environment.</td>
<td><img width="60%" src="http://localhost:4000/images/trainings-galaxy.png" alt="" />
</td>
</tr>
</table>
<p>We will perform a transcriptome-based mapping and estimates of transcript levels using Kallisto, and a differential analysis using EdgeR.</p>
<ul>
  <li>Connect to <a href="http://bioinfo-inter.ird.fr:8080/">Galaxy IRD</a></li>
  <li>Create a new history and import all the 8 RNASeq samples datasets (paired-end fastq files) from Data library
<code class="highlighter-rouge">Shared Data =&gt; Data Libraries =&gt; Galaxy_trainings_2019 =&gt; RNASeq</code></li>
  <li>Upload the Chr1 of rice transcriptome (cDNA) to be used as reference  - <code class="highlighter-rouge">http://rice.plantbiology.msu.edu/pub/data/Eukaryotic_Projects/o_sativa/annotation_dbs/pseudomolecules/version_7.0/chr01.dir/Chr1.cdna</code></li>
  <li>Run the kallisto quant program by providing Chr1 as transcriptome reference and specifying correctly pairs of input fastq- <code class="highlighter-rouge">kallisto quant</code>
You can do it with the pairs made one by one manually or you can make lists of dataset pairs. If you choose this second option:</li>
  <li>Build one list with the pairs of condition 1 and on other list with the pairs of condition 2.</li>
  <li>launch kallisto on each of the two lists =&gt; you get 2 kallisto outputs collections</li>
  <li>Convert kallisto outputs (collection of count files) into one single file that can be used as input for EdgeR -<code class="highlighter-rouge">Kallisto2EdgeR</code></li>
</ul>

<hr />

<p><a name="practice-2"></a></p>
<h3 id="practice-2--mapping-against-annotated-genome-reference-with-hisat2--counting-with-stringtie">Practice 2 : Mapping against annotated genome reference with Hisat2 + counting with Stringtie</h3>
<p><img width="20%" src="http://localhost:4000/images/toggleLogo2.png" alt="" /></p>

<h4 id="running-hisat2-and-stringtie-with-toggle">Running Hisat2 and Stringtie with TOGGLe</h4>

<p>Connection to account in IRD i-Trop cluster <code class="highlighter-rouge">ssh formationX@bioinfo-master.ird.fr</code></p>

<p>Input data are accessible from :</p>
<ul>
  <li>Input data : /data2/formation/tp-toggle/RNASeqData/</li>
  <li>Reference : /data2/formation/tp-toggle/RNASeqData/referenceFiles/chr1.fasta</li>
  <li>Annotation : /data2/formation/tp-toggle/RNASeqData/referenceFiles/chr1.gff3</li>
  <li>Config file: <a href="https://raw.githubusercontent.com/SouthGreenPlatform/TOGGLE/master/exampleConfigs/RNASeqHisat2Stringtie.config.txt">RNASeqReadCount.config.txt</a></li>
</ul>

<p>Import data from nas to your $HOME</p>
<ul>
  <li>Create a toggleTP directory in your $HOME <code class="highlighter-rouge">mkdir ~/TP-RNASEQ</code></li>
  <li>Make à copy for reference and input data into TP-RNASEQ directory: <code class="highlighter-rouge">cp -r /data2/formation/tp-toggle/RNASeqData/ ~/TP-RNASEQ</code></li>
  <li>Add the configuration file used by TOGGLe <code class="highlighter-rouge">cd ~/TP-RNASEQ/; wget https://raw.githubusercontent.com/SouthGreenPlatform/TOGGLE/master/exampleConfigs/RNASeqHisat2Stringtie.config.txt</code></li>
  <li>change SGE key <code class="highlighter-rouge">$sge</code> as below using a texte editor like nano <code class="highlighter-rouge">nano RNASeqHisat2Stringtie.config.txt</code></li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">$sge</span>
<span class="nt">-q</span> formation.q
<span class="nt">-b</span> Y
<span class="nt">-cwd</span></code></pre></figure>

<ul>
  <li>in TOGGLe configuration file use /scratch in <code class="highlighter-rouge">$scp</code> key to launch your job from scratch folder and also <code class="highlighter-rouge">$env</code> key using
<code class="highlighter-rouge">module load bioinfo/TOGGLE-dev/0.3.7</code> module installed on cluster.</li>
  <li>Check parameters of every step in <code class="highlighter-rouge">~/toggleTP/RNASeqData/RNASeqHisat2Stringtie.config.txt</code> as recommended by https://www.nature.com/articles/nprot.2016.095.</li>
</ul>

<p>Mapping is performed using HISAT2 and usually the first step, prior to mapping, is to create an index of the reference genome. TOGGle index genome automatically if indexes are absents in reference folder. 
In order to save some space think to only keep sorted BAM files one these are created.
After mapping, assemble the mapped reads into transcripts. StringTie can assemble transcripts with or without annotation, annotation can be helpful when the number of reads for a transcript is too low for an accurate assembly.</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">$order</span>
<span class="nv">1</span><span class="o">=</span>fastqc
<span class="nv">2</span><span class="o">=</span>hisat2
<span class="nv">3</span><span class="o">=</span>samtoolsView
<span class="nv">4</span><span class="o">=</span>samtoolsSort
<span class="nv">5</span><span class="o">=</span>stringtie 1
<span class="nv">1000</span><span class="o">=</span>stringtie 2

<span class="nv">$samtoolsview</span>
<span class="nt">-b</span>
<span class="nt">-h</span>

<span class="nv">$samtoolssort</span>

<span class="nv">$cleaner</span>
3

<span class="c">#PUT YOUR OWN SGE CONFIGURATION HERE</span>
<span class="nv">$sge</span>
<span class="nt">-q</span> formation.q
<span class="nt">-b</span> Y

<span class="nv">$stringtie</span> 1

<span class="nv">$stringtie</span> 2
<span class="nt">--merge</span>

<span class="nv">$hisat2</span>
<span class="nt">--dta</span>

<span class="nv">$scp</span>
/scratch/

<span class="nv">$env</span>
module load bioinfo/TOGGLE-dev/0.3.7</code></pre></figure>

<p>… Your data are now in ~/TP-RNASEQ.</p>

<p>Now, create a <code class="highlighter-rouge">runTOGGLeRNASEQ.sh</code> bash script to launch TOGGLe as follow :</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c">#!/bin/bash</span>
<span class="c">#$ -N TOGGLeRNAseq</span>
<span class="c">#$ -b yes</span>
<span class="c">#$ -q formation.q</span>
<span class="c">#$ -cwd</span>
<span class="c">#$ -V</span>

<span class="nv">dir</span><span class="o">=</span><span class="s2">"~/TP-RNASEQ/RNASeqData/fastq"</span>
<span class="nv">out</span><span class="o">=</span><span class="s2">"~/TP-RNASEQ/outTOGGLe"</span>
<span class="nv">config</span><span class="o">=</span><span class="s2">"~/TP-RNASEQ/RNASeqHisat2Stringtie.config.txt"</span>
<span class="nv">ref</span><span class="o">=</span><span class="s2">"~/TP-RNASEQ//RNASeqData/referenceFiles/chr1.fasta"</span>
<span class="nv">gff</span><span class="o">=</span><span class="s2">"~/TP-RNASEQ//RNASeqData/referenceFiles/chr1.gff3"</span>
<span class="c">## Software-specific settings exported to user environment</span>
module load bioinfo/TOGGLE-dev/0.3.7

<span class="c">#running tooglegenerator </span>
toggleGenerator.pl <span class="nt">-d</span> <span class="nv">$dir</span> <span class="nt">-c</span> <span class="nv">$config</span> <span class="nt">-o</span> <span class="nv">$out</span> <span class="nt">-r</span> <span class="nv">$ref</span> <span class="nt">-g</span> <span class="nv">$gff</span> <span class="nt">--report</span> <span class="nt">--nocheck</span><span class="p">;</span>

<span class="nb">echo</span> <span class="s2">"FIN de TOGGLe ^^"</span></code></pre></figure>

<p>Convert runTOGGLeRNASEQ in an executable file with <code class="highlighter-rouge">chmod +x runTOGGLeRNASEQ.sh</code></p>

<p>Launch runTOGGLeRNASEQ.sh in qsub mode</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">qsub <span class="nt">-q</span> formation.q <span class="nt">-N</span> TOGGLeRNASEQ <span class="nt">-b</span> yes <span class="nt">-cwd</span> <span class="s1">'module load bioinfo/TOGGLE-dev/0.3.7; ./runTOGGLeRNASEQ.sh '</span></code></pre></figure>

<p>Explore output <code class="highlighter-rouge">outTOGGLe</code> TOGGLe and check if everything was ok.</p>

<p>Open and explore<code class="highlighter-rouge">outTOGGLe/finalResults/intermediateResults.STRINGTIEMERGE.gtf</code></p>

<h3 id="now-that-we-have-our-assembled-transcripts-we-can-estimate-their-abundances">Now that we have our assembled transcripts, we can estimate their abundances.</h3>

<p>To estimate abundances, we have to run again stringtie using options -B and -e.</p>

<h5 id="prepare-data">prepare data</h5>

<ul>
  <li>Create symbolics links to order data before transferring them to <code class="highlighter-rouge">/scratch</code></li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nv">MOI</span><span class="o">=</span><span class="s2">"formationX"</span>
<span class="nv">OUTPUT</span><span class="o">=</span><span class="s2">"/home/</span><span class="nv">$MOI</span><span class="s2">/TP-RNASEQ/outTOGGLe/"</span>
mkdir <span class="nv">$OUTPUT</span>/stringtieEB
<span class="nb">cd</span> <span class="nv">$OUTPUT</span>/stringtieEB 
ln <span class="nt">-s</span> <span class="nv">$OUTPUT</span>/finalResults/intermediateResults.STRINGTIEMERGE.gtf <span class="nb">.</span>
ln <span class="nt">-s</span> <span class="nv">$OUTPUT</span>/output/<span class="k">*</span>/4_samToolsSort/<span class="k">*</span>SAMTOOLSSORT.bam .</code></pre></figure>

<h5 id="transfert-to-scratch">transfert to /scratch</h5>

<ul>
  <li>Remember good practices to work at IRD Cluster. <em>You have to copy data into a path /scratch in a node</em>. What is your node number?</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">qrsh <span class="nt">-q</span> formation.q
<span class="nv">MOI</span><span class="o">=</span><span class="s2">"formationX"</span>
<span class="nv">OUTPUT</span><span class="o">=</span><span class="s2">"/home/</span><span class="nv">$MOI</span><span class="s2">/TP-RNASEQ/outTOGGLe/"</span>
scp <span class="nt">-r</span> nas:<span class="nv">$OUTPUT</span>/stringtieEB /scratch/<span class="nv">$MOI</span> 
<span class="nb">cd</span> /scratch/<span class="nv">$MOI</span></code></pre></figure>

<h5 id="recovery-annotations">Recovery annotations</h5>

<ul>
  <li>Before merging gtf files obtained by stringtie, we have to recover the annotations in order to see the known genes names in gtf file. Stringtie annotate transcripts using gene id ‘MSTRG.1’ nomenclature . See https://github.com/gpertea/stringtie/issues/179</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">module load system/python/3.6.5
python3 /data2/formation/TP_RNA-seq_2019/gpertea-scripts/mstrg_prep.py intermediateResults.STRINGTIEMERGE.gtf <span class="o">&gt;</span> intermediateResults.STRINGTIEMERGE_prep.gtf</code></pre></figure>

<ul>
  <li>Compare gtf files before and after running <code class="highlighter-rouge">mstrg_prep.py</code> script. To do it you can choose a gene and explore differences:</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">grep</span> <span class="s1">'LOC_Os01g01010.1'</span> intermediateResults.STRINGTIEMERGE<span class="k">*</span></code></pre></figure>

<h5 id="gffcompare">gffcompare</h5>

<ul>
  <li>Let’s compare the StringTie transcripts to known transcripts using gffcompare https://github.com/gpertea/gffcompare and explore results. Observe statistics. How many “J”, “U” and “=” do you obtain?. <code class="highlighter-rouge">gffcompare_out.annotated.gtf</code> file will be visualised with IGV later.</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">scp ~/TP-RNASEQ//RNASeqData/referenceFiles/chr1.fasta <span class="nb">.</span>
/data2/formation/TP_RNA-seq_2019/gffcompare/gffcompare <span class="nt">-r</span> chr1.gff3 <span class="nt">-o</span> gffcompare_out  intermediateResults.STRINGTIEMERGE_prep.gtf</code></pre></figure>

<h5 id="stringtie--e--b">Stringtie -e -B</h5>

<ul>
  <li>… Finally, we launch stringtie: (change nodeXX by your node number)</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for </span>i <span class="k">in</span> <span class="k">*</span>bam <span class="p">;</span> <span class="k">do </span><span class="nb">eval</span> <span class="s2">"mkdir </span><span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.bam/</span><span class="k">}</span><span class="s2">; qsub -q formation.q@nodeXX -N stringtie2 -cwd -V -b yes 'module load bioinfo/stringtie/1.3.4; stringtie"</span> <span class="nv">$PWD</span><span class="s2">"/"</span><span class="nv">$i</span> <span class="s2">"-G </span><span class="nv">$PWD</span><span class="s2">"</span>/<span class="s2">"intermediateResults.STRINGTIEMERGE_prep.gtf -e -B -o </span><span class="nv">$PWD</span><span class="s2">/</span><span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.bam/</span><span class="k">}</span><span class="s2">/</span><span class="k">${</span><span class="nv">i</span><span class="p">/bam/count</span><span class="k">}</span><span class="s2">'"</span><span class="p">;</span> <span class="k">done</span></code></pre></figure>

<h5 id="convert-to-counts-table">Convert to counts table</h5>

<ul>
  <li>Convert stringtie output in counts using <code class="highlighter-rouge">prepDE.py</code>. Dont forget! You are in /scratch <code class="highlighter-rouge">/scratch/formationX</code></li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">mkdir counts
<span class="nb">cd </span>counts/
ln <span class="nt">-s</span> /scratch/<span class="nv">$MOI</span>/<span class="k">*</span>/<span class="k">*</span>.count <span class="nb">.</span>
<span class="k">for </span>i <span class="k">in</span> <span class="k">*</span>.count<span class="p">;</span> <span class="k">do </span><span class="nb">echo</span> <span class="k">${</span><span class="nv">i</span><span class="p">/.SAMTOOLSSORT.count/</span><span class="k">}</span> <span class="nv">$PWD</span>/<span class="nv">$i</span><span class="p">;</span> <span class="k">done</span> <span class="o">&gt;</span> listGTF.txt
python2 /data2/formation/TP_RNA-seq_2019/prepDE.py <span class="nt">-i</span> listGTF.txt</code></pre></figure>

<p>You have obtained <code class="highlighter-rouge">gene_count_matrix.csv</code> and <code class="highlighter-rouge">transcript_count_matrix.csv</code></p>

<h5 id="transfer-data-from-scratch-to-your-home-on-cluster">Transfer data from /scratch to your home on cluster</h5>

<ul>
  <li>Don’t forget scp *.counts files to your $OUTPUT</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">scp <span class="nt">-r</span> /scratch/<span class="nv">$MOI</span>/counts/ ~/TP-RNASEQ/
scp <span class="nt">-r</span> /scratch/<span class="nv">$MOI</span>/gffcompare<span class="k">*</span>/ ~/TP-RNASEQ/</code></pre></figure>

<h5 id="transfer-data-to-local-machine">Transfer data to local machine</h5>

<ul>
  <li>From your local terminal, transfer counts to your local machine with scp</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">scp <span class="nt">-r</span> formationX@bioinfo-nas.ird.fr:/home/formationX/TP-RNASEQ/counts/ .</code></pre></figure>

<ul>
  <li>Transfer also reference files fasta, gff and <code class="highlighter-rouge">gffcompare_out.annotated.gtf</code> to use it later with IGV.</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">scp <span class="nt">-r</span> formationX@bioinfo-nas.ird.fr:/home/formationX/toggleTP/RNASeqData/referenceFiles/<span class="k">*</span>.gff3 <span class="nb">.</span>
scp <span class="nt">-r</span> formationX@bioinfo-nas.ird.fr:/home/formationX/toggleTP/RNASeqData/referenceFiles/<span class="k">*</span>.fasta <span class="nb">.</span>
scp <span class="nt">-r</span> formationX@bioinfo-nas.ird.fr:/home/formationX/TP-RNASEQ/gffcompare<span class="k">*</span> .</code></pre></figure>

<hr />

<p><a name="practice-3"></a></p>
<h3 id="practice-3--differential-expression-analysis-using-edger-and-deseq2">Practice 3 : Differential expression analysis using EdgeR and DESeq2</h3>
<td>Practice 3 will be performed in PIVOT via R Studio.</td>

<p>PIVOT: Platform for Interactive analysis and Visualization Of Transcriptomics data
Qin Zhu, Junhyong Kim Lab, University of Pennsylvania
Oct 26th, 2017</p>

<h4 id="intallation-of-pivot">Intallation of PIVOT</h4>

<p>Dependencies that needs to be manually installed.
You may need to paste the following code line by line 
and choose if previously installed packages should be updated (recommended).</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">install.packages<span class="o">(</span><span class="s2">"devtools"</span><span class="o">)</span> 
library<span class="o">(</span><span class="s2">"devtools"</span><span class="o">)</span>
install.packages<span class="o">(</span><span class="s2">"BiocManager"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"BiocUpgrade"</span><span class="o">)</span> 
<span class="c">#BiocManager::install("GO.db")</span>
BiocManager::install<span class="o">(</span><span class="s2">"HSMMSingleCell"</span><span class="o">)</span>
<span class="c">#BiocManager::install("org.Mm.eg.db")</span>
<span class="c">#BiocManager::install("org.Hs.eg.db")</span>
BiocManager::install<span class="o">(</span><span class="s2">"DESeq2"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"SingleCellExperiment"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"scater"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"monocle"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"GenomeInfoDb"</span><span class="o">)</span></code></pre></figure>

<h4 id="install-pivot">Install PIVOT</h4>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">install_github<span class="o">(</span><span class="s2">"qinzhu/PIVOT"</span><span class="o">)</span>
BiocManager::install<span class="o">(</span><span class="s2">"BiocGenerics"</span><span class="o">)</span> <span class="c"># You need the latest BiocGenerics &gt;=0.23.3</span></code></pre></figure>

<h3 id="launch-pivot">Launch PIVOT</h3>

<p>To run PIVOT, in Rstudio console related to a web shiny interface, use command</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">library<span class="o">(</span>PIVOT<span class="o">)</span>
pivot<span class="o">()</span></code></pre></figure>

<p>Go to your web browser.</p>

<hr />

<h3 id="introduction">Introduction</h3>

<p>As you have seen, RNA sequencing (RNA-seq) is an experimental method for measuring RNA expression levels via high-throughput sequencing of small adapter-ligated fragments.
The number of reads that map to a gene is the proxy for its expression level.
There are many steps between receiving the raw reads from the sequencer and gaining biological insight.
In this tutorial, we will start with a “Table of counts” and end with a “List of differentially expressed genes”.</p>

<h4 id="step-1---data-input">Step 1  : Data Input</h4>

<p>To input expression matrix, select “Counts Table” as input file type. PIVOT expects the count matrix to have rows as genes and samples as columns.
Gene names and sample names should be the first column and the first row, respectively.</p>

<ul>
  <li>Go to file Tab.</li>
  <li>Take the count file <code class="highlighter-rouge">gene_count_matrix.csv</code> generated previously.</li>
  <li>Import this file into  Data input  and then Input file type.</li>
  <li>Add Use Tab separator to Skip Rows.</li>
  <li>Check if yours data are imported in the rigth window.</li>
</ul>

<h4 id="step-2--input-design-information">Step 2 : Input Design Information</h4>

<p>The design infomation are used for sample point coloring and differential expression analysis. Users can input the entire sample meta sheet as 
design information, or manually specify groups or batches for each sample.</p>

<ul>
  <li>Go to DESIGN.</li>
  <li>Go to Designed Table Upload. Upload <code class="highlighter-rouge">info.txt</code></li>
  <li>Verify that the header of the info file corresponds to the count file.</li>
  <li>Choose the Separator : Space or the appropriate separator.</li>
  <li>Verify on the Design Table Preview and submit design.</li>
  <li>Choose the Normalieation Method :</li>
  <li>for Edge R you can use <code class="highlighter-rouge">DESeq, Trimmed Mean of M-values TMM, or Upperquartile</code>.</li>
  <li>for DESeq you can use <code class="highlighter-rouge">DESeq, Modifed DESeq</code></li>
</ul>

<p>In order to have a quick view of your chosen data, look at the summary.</p>

<h4 id="step-3--feature-filtering">step 3 : Feature Filtering</h4>

<ul>
  <li>There are currently 3 types of feature filter in PIVOT: the expression filter, which filters based on various expression statistics;</li>
  <li>You can choose the filter criteria.</li>
</ul>

<h4 id="step-4--select-samples">Step 4 : Select samples</h4>

<ul>
  <li>Go To SAMPLE</li>
  <li>Select your sample and condition,
    <ul>
      <li>Subset by sample and or subset by list, as you can create different experiment analysis.</li>
    </ul>
  </li>
</ul>

<p>DATA MAP draw a summary of your different analysis, so you can save the history of your analysis.</p>

<h4 id="step-5--basic-statistics">Step 5 : Basic statistics</h4>

<ul>
  <li>If you want to keep the count table, upload it.</li>
  <li>Check the distribution of each condition in the standard deviation graph, the dispersion graph.</li>
  <li>If needed, you can download the Variably Expressed Genes, and on the graph, you can see the dispersion of your data.</li>
</ul>

<h4 id="first-part-with-edger-on-pivot">First part with EdgeR on PIVOT</h4>

<ul>
  <li>Run the EdgeR program for differential analysis - <code class="highlighter-rouge">EdgeR</code></li>
  <li>Check relevance of normalized expression values provided by EdgeR</li>
  <li>Observe MDS plot of experimental conditions. Observe Smear plot.</li>
</ul>

<p>Questions :</p>
<ul>
  <li>Using filters parameters, determine how many genes are found to be differentally expressed using a minimum pvalue &lt;= 0.05, 0.1? Using a minimum FDR-adjusted pvalue &lt;= 0.05, 0.1?</li>
</ul>

<h4 id="step-6---differential-expression-with-edger">Step 6  : Differential Expression with EdgeR</h4>

<p>Once data have been normalized in the Step 1, you can choose the method to find the Differential expression gene between the condition previously choosen. 
In edgeR, it is recommended to remove genes with very low counts.  Remove genes (rows) which have zero counts for all samples from the dge data object.</p>

<ul>
  <li>Go to Differential Expression, choose edgeR</li>
  <li>According to your normalized method choice in step 1, notify the Data Normalization method, and choose Experiment Design, and condition.</li>
  <li>For this dataset choose <code class="highlighter-rouge">condition, condition</code></li>
  <li>Choose the Test Method, Exact test, GLM likelihood ratio test, and GLM quasi-likelihood F-test.</li>
  <li>For this dataset choose <code class="highlighter-rouge">Exact test</code></li>
</ul>

<p>Then <code class="highlighter-rouge">Run Modeling</code></p>

<ul>
  <li>Choose the P adjustement adapted</li>
  <li>For this dataset choose <code class="highlighter-rouge">False discovery rate</code> or <code class="highlighter-rouge">Bonferroni correction</code></li>
  <li>Choose FDR cutoff 0.1 and 0.05</li>
  <li>Choose Term 1 and Term 2</li>
</ul>

<p>You obtain a table containing the list of the differential gene expressed according to your designed analysis. This table contains logFoldChange, logCountPerMillion, PValue, FDR. 
This table can be download in order to use it for other analysis. The Mean-Difference Plot show you the up-regulated gene in green, and Down regulated gene in black.</p>

<ul>
  <li>Keep this file <code class="highlighter-rouge">edgeR_results.csv</code></li>
  <li>Change the name of the file according to your analysis.</li>
  <li>Example : <code class="highlighter-rouge">edgeR_results_FDR_01.csv</code> to be able to recognize the different criteria used.</li>
</ul>

<p>Questions :</p>
<ul>
  <li>Do you have the sample number of differential gene for a FDR cutoff 0.1 and 0.05?</li>
  <li>According to you, what is the best cutoff?</li>
  <li>When you change the order of the Term1 and Term2, what are the consequences on the Results Table?</li>
</ul>

<hr />

<h4 id="second-part-with-deseq2-on-pivot">Second part with DESeq2 on PIVOT</h4>

<ul>
  <li>Run the DESeq program for differential analysis - <code class="highlighter-rouge">DESeq2</code></li>
  <li>Check relevance of normalized expression values provided by DESeq2</li>
  <li>Observe MDS plot of experimental conditions. Observe Smear plot.</li>
</ul>

<p>Questions:</p>
<ul>
  <li>Using filters parameters, determine how many genes are found to be differentially expressed using a minimum pvalue &lt;= 0.05, 0.1? Using a minimum FDR-adjusted pvalue &lt;= 0.05, 0.1?</li>
</ul>

<hr />

<h4 id="step-6---differential-expression-with-deseq2">Step 6  : Differential Expression with DESeq2</h4>

<p>Once data have been normalized in the Step 1, you can choose the method to find the Differential expression gene between the condition previously choosen.</p>

<ul>
  <li>Go to Differential Expression, choose DEseq2</li>
  <li>According to your normalized method choice in step 1, notify the Data Normalization method, and choose Experiment Design, and condition.</li>
  <li>For this dataset choose <code class="highlighter-rouge">condition, condition</code></li>
  <li>Choose the Test Method, Exact test, GLM likelihood ratio test, and GLM quasi-likelihood F-test.</li>
  <li>For this dataset choose <code class="highlighter-rouge">Exact test</code></li>
</ul>

<p>Then <code class="highlighter-rouge">Run Modeling</code></p>

<ul>
  <li>Choose the P adjustement adapted</li>
  <li>For this dataset choose <code class="highlighter-rouge">FDR cutoff</code></li>
  <li>Choose FDR cutoff 0.1 and 0.05</li>
</ul>

<p>You obtain a table containing the list of the differential gene expressed according to your designed analysis. This table contains baseMean, log2FoldChange, PValue, pvalueadjust. 
This table can be downloaded in order to use it for other analysis. The MA Plot show you the up-regulated gene LFC&gt;0, and Down regulated gene LFC&lt;0, the differential gene are in red.</p>

<ul>
  <li>Keep this file <code class="highlighter-rouge">deseq_results.csv</code></li>
  <li>Change the name of the file according to your analysis.</li>
  <li>Example : <code class="highlighter-rouge">deseq_results_FDR_01.csv</code> to be able to recognize the different criteria used.</li>
</ul>

<p>Questions :</p>
<ul>
  <li>Do you have the sample number of defferential gene for a FDR cutoff 0.1 and 0.05?</li>
  <li>According to you, what is the best cutoff?</li>
  <li>When you change the order of the Term1 and Term2, what are the consequence on the Results Table?</li>
</ul>

<hr />

<h4 id="step-7---visualisation-heatmap-correlation-pca">Step 7  : Visualisation Heatmap, Correlation, PCA</h4>

<p>You can make correlation between the control and the rep in order to identify library bias if exists.
We can also explore the relationships between the samples by visualizing a heatmap of the correlation matrix.
The heatmap result corresponds to what we know about the data set.
First, the samples in group 1 and 2 come from the control and the repetition, so the two groups are very different.
Second, since the samples in each group are technical replicates, the within group variance is very low.</p>

<h4 id="correlation-between-sample-and-control">Correlation between sample and control</h4>
<ul>
  <li>Go to Correlation.</li>
  <li>1 Pairwaise Scatterplot show the pairwaise comparison between your samples.</li>
  <li>2 Sample correlation with 3 methods, pearson, sperman or kendal, with the agglomeration method show you how are linked your samples.</li>
  <li>3 Feature Correlation represented with a heatMap ordered by variance of the expression.</li>
</ul>

<h4 id="pca">PCA</h4>
<ul>
  <li>To separate your sample a PCA is  way to make a dimension reduction.</li>
</ul>

<p>Question :</p>
<ul>
  <li>Can you describe the structure of your sample, with a correlation analysis, and a PCA?</li>
</ul>

<hr />

<p><a name="practice-4"></a></p>
<h3 id="practice-4--compare-list-of-de-genes-with-edger-and-deseq2">Practice 4 : Compare list of DE genes with EdgeR and DESeq2</h3>
<p>Practice 4 will be performed with Venn Diagramm implemented on PIVOT.</p>

<ul>
  <li>Compare lists of DE genes with the two approches.</li>
  <li>Go to Toolkit.</li>
  <li>Upload your lists of gene obtained with edegR <code class="highlighter-rouge">edgeR_results_FDR_01.csv</code> , and DESeq2 <code class="highlighter-rouge">deseq_results_FDR_01.csv</code>.</li>
  <li>Upload the first list then Add List and upload the second list.</li>
  <li>You can see the Venn diagram and download the common list of gene between the both methods.</li>
</ul>

<p>Questions:</p>

<ul>
  <li>Look at the expression values for a gene found DE with EdgeR and not with DESeq2, and vice-versa, give the pvalue of each gene?</li>
</ul>

<p>Some other tools are available to compare 2 lists of gene. <a href="http://bioinfogp.cnb.csic.es/tools/venny/">Venny</a>.</p>

<hr />

<p><a name="practice-5"></a></p>
<h3 id="practice-5--hierarchical-clustering">Practice 5 : Hierarchical Clustering</h3>
<p>Practice 5 will be performed with PIVOT.</p>
<ul>
  <li>Connect to your PIVOT interface.</li>
  <li>Go to Clustering.</li>
  <li>For each analysis EdgeR or DESeq2 specify the Data Input (count, log…).</li>
  <li>Choose the distance <code class="highlighter-rouge">Euclidean</code> or an other, the Agglomeration method <code class="highlighter-rouge">Ward</code>and the number of cluster.</li>
</ul>

<hr />

<p><a name="practice-6"></a></p>
<h3 id="practice-6--visualization-of-mapped-reads-against-genes-using-igv">Practice 6 : Visualization of mapped reads against genes using IGV</h3>
<p>Practice 6 will be performed with Integrated Genome Viewer (IGV).</p>

<ul>
  <li>
    <p>Focus on a gene that has been found to be differentially expressed and observe the structure of the gene.</p>
  </li>
  <li>
    <p>From master0 <code class="highlighter-rouge">qlogin -q formation.q</code></p>
  </li>
  <li>
    <p>Lauch <code class="highlighter-rouge">samtools index</code> using bam obtained by hisat2:</p>
  </li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for </span>fl <span class="k">in</span> ./<span class="k">*</span>.bam<span class="p">;</span> <span class="k">do </span>samtools index <span class="nv">$fl</span><span class="p">;</span> <span class="k">done</span></code></pre></figure>

<ul>
  <li>
    <p>Run igv : <code class="highlighter-rouge">igv.sh &amp;</code></p>
  </li>
  <li>
    <p>Load reference genome, GFF annotation file, BAMs files and the gffCompare <code class="highlighter-rouge">gffcompare_out.annotated.gtf</code> output.</p>
  </li>
  <li>
    <p>Recovery some ID to visualise it in IGV:</p>
  </li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">grep</span> <span class="s1">'class_code "u"'</span> gffcompare_out.annotated.gtf | less</code></pre></figure>

<ul>
  <li>
    <p>Copy a identifiant, for example, “XLOC_000469” et and search it in IGV. Show this loci.</p>
  </li>
  <li>
    <p>or  :</p>
  </li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="nb">grep</span> <span class="s1">'class_code "x"'</span> gffcompare_out.annotated.gtf | less</code></pre></figure>

<ul>
  <li>Search other gene, for example, “LOC_Os01g01710”.</li>
</ul>

<hr />

<h3 id="links">Links</h3>
<p><a name="links"></a></p>

<ul>
  <li>Related courses : <a href="https://southgreenplatform.github.io/trainings/linuxJedi/">Transcriptomics</a></li>
  <li>Degust : <a href="http://degust.erc.monash.edu/">Degust</a></li>
  <li>MeV: <a href="http://mev.tm4.org/">MeV</a></li>
  <li>MicroScope: <a href="http://microscopebioinformatics.org/">MicroScope</a></li>
  <li>Comparison of methods for differential expression: <a href="https://southgreenplatform.github.io/trainings//files/Comparison_of_methods_for_differential_gene_expression_using RNA-seq_data.pdf">Report</a></li>
  <li>PIVOT: <a href="https://github.com/qinzhu/PIVOT/">PIVOTGithub</a></li>
  <li>DESeq2: <a href="http://www.bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a></li>
  <li>EdgR: <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">EdgeR</a></li>
</ul>

<hr />

<h3 id="license">License</h3>
<p><a name="license"></a></p>

<div>
The resource material is licensed under the Creative Commons Attribution 4.0 International License (<a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">here</a>).
<center><img width="25%" class="img-responsive" src="http://creativecommons.org.nz/wp-content/uploads/2012/05/by-nc-sa1.png" />
</center>
</div>


	</div> <!-- #main -->
</div> <!-- #main-container -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="http://localhost:4000/js/jquery-1.11.3.min.js"></script>
    <script src="http://localhost:4000/js/bootstrap.min.js"></script>
    <script src="http://localhost:4000/js/homepage.js"></script>
    <script src="http://localhost:4000/js/libs/modernizr-2.0.6.min.js"></script>

</script>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <div id="footer">
			<div class="logo-div" class="clearfix">
				<a class="logo" href="http://www.ird.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo_ird.png" alt="IRD Logo"  width="" />
				</a>

				<a class="logo" href="http://www.cirad.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo-cirad.png" alt="CIRAD Logo"   width="" />
				</a>

				<a class="logo" href="http://www.inra.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo_inra.png" alt="INRA Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.bioversityinternational.org/">
					<img class="img-logo" src="http://localhost:4000/images/logo_bioversity.png" alt="Bioversity Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.supagro.fr/">
					<img class="img-logo" src="http://localhost:4000/images/logo_supagro.png" alt="SupAgro Logo"   width=""/>
				</a>
				
				<a class="logo" href="http://www.southgreen.fr">
					<img class="img-logo" src="http://localhost:4000/images/southgreenlong.png" alt="South Green Logo"   width=""/>
				</a>

			</div>
		</div>

		<div class="footer_txt">
			<p>The contents of this website are &copy;&nbsp;2019 under the terms of the Licencied under
				<a href="http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html">CeCill-C</a>
				 and <a href="/blob/master/LICENSE">GPLv3</a>.
			</p>
            <p class="webmaster">WebMaster: <a class="webmaster" target="_blank" href="https://d2gg9evh47fn9z.cloudfront.net/800px_COLOURBOX6415001.jpg">Christine</a>
            &
            <a class="webmaster" target="_blank" href="http://moziru.com/images/fluffy-clipart-silly-monster-13.png">Séb</a></p>
		</div>
      </footer>
    </div>

</body>
</html>
