<!doctype html>
<!--[if lt IE 7]><html class="no-js ie6 oldie" lang="ja"><![endif]-->
<!--[if IE 7]>   <html class="no-js ie7 oldie" lang="ja"><![endif]-->
<!--[if IE 8]>   <html class="no-js ie8 oldie" lang="ja"><![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="ja"><!--<![endif]-->
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">

    <title>Advanced Linux Practice</title>
    <meta name="description" content="Welcome to Galaxy South Green">
    <meta name="description" content="Advanced Linux Practice page">
    
    <meta name="author" content=" Marilyne Summo ">
    

    <meta name="viewport" content="width=device-width,initial-scale=1">

    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/bootstrap.min.css" >
    <link rel="stylesheet" type="text/css" media="screen" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/monokai.css">
    <link rel="stylesheet" type="text/css" media="screen" href="http://localhost:4000/css/stylesheet.css">
    <link rel="icon" href="http://localhost:4000/images/southgreenlong.png">
</head>
 <body>

      <!-- Static navbar -->
      <nav class="navbar navbar-default navbar-inverse navbar-static-top" style="margin-bottom:0px">
        <div class="container">
          <div class="navbar-header">
            <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false" aria-controls="navbar">
              <span class="sr-only"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
              <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand visible-xs" href="#"></a>
          </div>
          <div id="navbar" class="navbar-collapse collapse">
			<ul id="menu">
        <li><a href="http://localhost:4000/">Home</a></li>
        <li><a href="http://localhost:4000/training_home">Trainings</a></li>
								<li><a href="http://localhost:4000/contact/">About Us</a></li>
			</ul>
          </div>
        </div>
      </nav>
      <!--/.nav-collapse -->
    <!-- Header -->
    <div id="header_galaxy" class="outer">
        <header class="inner">
          <div class="header">
            <img width="40%" src="http://localhost:4000/images/southgreenlong.png" alt="South Green Logo" />
            <h2 id="project_tagline">Welcome to <img style="width: 26px; vertical-align: top;" src="http://localhost:4000/images/galaxyIcon_noText.png"> Galaxy</h2>
          </div>
        </header>
    </div>

<body>
<div id="main_content_wrap" class="outer">
	 <section id="main_content" class="inner">
<table>
  <thead>
    <tr>
      <th style="text-align: left">Description</th>
      <th style="text-align: left">Hands On Lab Exercises for Linux</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Related-course materials</td>
      <td style="text-align: left"><a href="https://southgreenplatform.github.io/trainings/linux/linuxPracticeJedi//">Linux for Jedi</a></td>
    </tr>
    <tr>
      <td style="text-align: left">Authors</td>
      <td style="text-align: left">Christine Tranchant-Dubreuil (christine.tranchant@ird.fr) &amp; Gautier Sarah (gautier.sarah</td>
    </tr>
    <tr>
      <td style="text-align: left">Creation Date</td>
      <td style="text-align: left">11/03/2018</td>
    </tr>
    <tr>
      <td style="text-align: left">Last Modified Date</td>
      <td style="text-align: left">14/04/2019</td>
    </tr>
  </tbody>
</table>

<hr />

<h3 id="summary">Summary</h3>

<!-- TOC depthFrom:2 depthTo:2 withLinks:1 updateOnSave:1 orderedList:0 -->
<ul>
  <li><a href="#preambule">Preambule: Softwares to install before connecting to a distant linux server </a></li>
  <li><a href="#practice-1">Practice 1: Get Connecting on a linux server by <code class="highlighter-rouge">ssh</code></a></li>
  <li><a href="#practice-2">Practice 2: Preparing working environnement</a></li>
  <li><a href="#practice-3">Practice 3: Monitoring processes) with <code class="highlighter-rouge">w, ps, kill, top</code></a></li>
  <li><a href="#practice-4">Practice 4: Using the <code class="highlighter-rouge">&amp;&amp;</code> separator</a></li>
  <li><a href="#practice-5">Practice 5: Searching for text using <code class="highlighter-rouge">grep</code></a></li>
  <li><a href="#practice-6">Practice 6: Displaying lines with <code class="highlighter-rouge">sed</code></a></li>
  <li><a href="#practice-7">Practice 7: Deleting lines with <code class="highlighter-rouge">sed</code></a></li>
  <li><a href="#practice-8">Practice 8: Parsing files with <code class="highlighter-rouge">sed</code> using regexp</a></li>
  <li><a href="#practice-9">Practice 9: Modifying files with <code class="highlighter-rouge">sed</code></a></li>
  <li>
    <p><a href="#practice-10">Practice 10: Manipulating files with <code class="highlighter-rouge">awk</code></a></p>
  </li>
  <li><a href="#links">Links</a></li>
  <li><a href="#license">License</a></li>
</ul>

<hr />

<p><a name="preambule"></a></p>
<h3 id="preambule">Preambule</h3>
<ul>
  <li>List of Softwares to install before connecting to a distant linux server <a href="https://southgreenplatform.github.io/trainings/linux/linuxPractice/#preambule">more information</a></li>
  <li>Arborescence image :</li>
</ul>

<hr />

<p><a name="practice-1"></a></p>
<h3 id="practice-1--get-connecting-on-a-linux-server-by-ssh">Practice 1 : Get Connecting on a linux server by <code class="highlighter-rouge">ssh</code></h3>

<p>In mobaXterm:</p>
<ol>
  <li>Click the session button, then click SSH.
    <ul>
      <li>In the remote host text box, type: HOSTNAME (see table below)</li>
      <li>Check the specify username box and enter your user name</li>
    </ul>
  </li>
  <li>In the console, enter the password when prompted.
Once you are successfully logged in, you will use this console for the rest of the lecture.</li>
</ol>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Cluster HPC</th>
      <th style="text-align: left">hostname</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">IRD HPC</td>
      <td style="text-align: left">bioinfo-master.ird.fr</td>
    </tr>
    <tr>
      <td style="text-align: left">AGAP HPC</td>
      <td style="text-align: left">cc2-login.cirad.fr</td>
    </tr>
  </tbody>
</table>

<ul>
  <li>Connect on the HPC</li>
</ul>

<hr />

<p><a name="practice-2"></a></p>
<h3 id="practice-2--preparing-working-environnement">Practice 2 : Preparing working environnement</h3>

<ul>
  <li>Type qrsh to connect on one node</li>
  <li>Move into the directory /scratch</li>
  <li>Create a working directory such as Formation-X (X corresponds to your login id/number)</li>
  <li>Move into this directory just created and check the current/working directory just by looking the prompt</li>
</ul>

<hr />

<p><a name="practice-3"></a></p>
<h3 id="practice-3---monitoring-processes">Practice 3 :  Monitoring processes</h3>

<h4 id="displaying-the-list-of-processes">Displaying the list of processes</h4>
<ul>
  <li>Type the command <code class="highlighter-rouge">w</code> through 2 consoles : one connected on bioinfo-master, the other connected on one node</li>
  <li>Type (on the node) the command <code class="highlighter-rouge">ps</code>without option, then with the option <code class="highlighter-rouge">u</code>, <code class="highlighter-rouge">ua</code>,  <code class="highlighter-rouge">uax</code></li>
  <li>Type the command <code class="highlighter-rouge">top</code>on the node</li>
  <li>Then use the “option” c to display the complete process</li>
  <li>Then use the “option” u to display only your processes</li>
</ul>

<h4 id="kill-a-process---downloading-files-from-sra-through-two-ways">Kill a process - downloading files from SRA through two ways</h4>
<p>We want to download one fastq file from NCBI SRA (available here https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR304976 ) using SRAtoolkit as below :</p>

<p><code class="highlighter-rouge">module load sratoolkit
fastq-dump SRRXXXX</code></p>

<p>This will download the SRA file (in sra format) and then convert them to fastq file for you. More details on https://isugenomics.github.io/bioinformatics-workbook/dataAcquisition/fileTransfer/sra.html</p>

<ul>
  <li>Download the fastq file in the directory created precedently in /scratch</li>
  <li>Open a new terminal to connect on bioinfo-master.ird.fr</li>
  <li>From this terminal, display the list of processes running on the node where you are downloading the fastq file with fastq-dump</li>
  <li>From this terminal, kill your process “fastq-dump” directly from bioinfo-master</li>
</ul>

<hr />

<p><a name="practice-4"></a></p>
<h3 id="practice-4--using-the--separator">Practice 4 : Using the &amp;&amp; separator</h3>

<ul>
  <li>On the console, type the 2 following linux commands to get data necessary for the next (we will explain the two commands latter):</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c"># get the file on the web and decompress the gzip file </span>
wget http://sg.ird.fr/LINUX-TP/LINUX-TP.tar.gz <span class="o">&amp;&amp;</span> <span class="nb">tar</span> <span class="nt">-xzvf</span> LINUX-TP.tar.gz</code></pre></figure>

<ul>
  <li>Check the content of your home directory on the server now (cf. filetree just below)</li>
</ul>

<p><img width="50%" class="img-responsive" src="http://localhost:4000/images/tpLinux/tp-arbo.png" /></p>

<hr />

<p><a name="practice-5"></a></p>
<h3 id="practice-5--searching-for-text-using-grep">Practice 5 : Searching for text using <code class="highlighter-rouge">grep</code></h3>

<h4 id="from-a-gff-file">from a gff file</h4>

<ul>
  <li>Go on the following page : http://rice.plantbiology.msu.edu/pub/data/Eukaryotic_Projects/o_sativa/annotation_dbs/pseudomolecules/version_7.0/</li>
  <li>Copy the url of the rice genome annotation file (gff format) that we will use to download the file directly on the server</li>
  <li>Go to the <code class="highlighter-rouge">bank</code> directory and type the following command :</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">wget gff_url</code></pre></figure>

<ul>
  <li>Prints the number of lines with the word <code class="highlighter-rouge">gene</code> in the gff file - <code class="highlighter-rouge">grep -P</code></li>
  <li>Counts the number of genes - <code class="highlighter-rouge">grep -c</code></li>
  <li>Search for the nbs-lrr genes - <code class="highlighter-rouge">grep -i</code></li>
  <li>Removes the lines with <code class="highlighter-rouge">putative</code> word - <code class="highlighter-rouge">grep -v</code></li>
  <li>Counts the number of mRNA in the chromosome 1 - <code class="highlighter-rouge">grep -c regexp</code></li>
  <li>Counts the number of mRNA in the first five chromosomes - <code class="highlighter-rouge">grep -c regexp</code></li>
</ul>

<h4 id="from-a-fasta-file">from a fasta file</h4>
<ul>
  <li>Get from the same website the cDNA sequences of the rice genome (fasta format) <code class="highlighter-rouge">wget</code></li>
  <li>Get the help of <code class="highlighter-rouge">infoseq</code> program - <code class="highlighter-rouge">infoseq --h</code></li>
  <li>Run infoseq program on the fasta file just downloaded</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"> infoseq <span class="nt">-sequence</span> FASTA_FILE | head</code></pre></figure>

<ul>
  <li>Display only accession, length and pgc column with the options of <code class="highlighter-rouge">infoseq</code></li>
  <li>What is the shorthest sequence (Accession and length)? <code class="highlighter-rouge">infoseq, sort, head</code></li>
  <li>What is the longuest sequence (Accession and length)? <code class="highlighter-rouge">infoseq, sort, head</code></li>
  <li>Count the number of sequences with a length between 1000 and 9999 with <code class="highlighter-rouge">grep</code></li>
</ul>

<hr />

<p><a name="practice-6"></a></p>
<h3 id="practice-6---displaying-lines-with-sed">Practice 6 :  Displaying lines with <code class="highlighter-rouge">sed</code></h3>
<p>For this exercise, you will work on the fastq file LINUX-TP/Data/fastq/pairedTwoIndividusGzippedIrigin/irigin1_1.fastq.gz</p>

<ul>
  <li>Print the 8 first lines</li>
  <li>Print the lines 5 to 12</li>
  <li>Print only the sequences ids</li>
  <li>Print only the sequences ids and nucleotides sequences</li>
</ul>

<hr />

<p><a name="practice-7"></a></p>
<h3 id="practice-7--deleting-lines-with-sed">Practice 7 : Deleting lines with <code class="highlighter-rouge">sed</code></h3>
<p>For this exercise, you will work on the fastq file LINUX-TP/Data/fastq/pairedTwoIndividusIrigin/irigin1_1.fastq</p>

<ul>
  <li>Delete the end of the file from the line 9</li>
  <li>Delete the lines containing only a <code class="highlighter-rouge">+</code></li>
  <li>Delete the lines containing only a <code class="highlighter-rouge">+</code> and the quality sequences</li>
</ul>

<hr />

<p><a name="practice-8"></a></p>
<h3 id="practice-8--file-parsing-with-sed-using-regexp">Practice 8 : File parsing with <code class="highlighter-rouge">sed</code> using regexp</h3>

<h4 id="from-the-gff-file-precedently-downloaded">From the gff file precedently downloaded</h4>
<ul>
  <li>Count the number of genes</li>
</ul>

<h4 id="from-a-vcf-file">From a vcf file</h4>
<ul>
  <li>Download the vcf file available at this url http://sg.ird.fr/LINUX-TP/OgOb-all-MSU7-CHR6.GATKVARIANTFILTRATION-100000.vcf.tar.gz</li>
  <li>How many polymorphisms were considered bad and filtered out (Displaying all the lines without neither the <code class="highlighter-rouge">PASS</code> tag nor starting with <code class="highlighter-rouge">#</code> )?</li>
</ul>

<hr />

<p><a name="practice-9"></a></p>
<h3 id="practice-9--file-modification-with-sed">Practice 9 : File modification with <code class="highlighter-rouge">sed</code></h3>

<h4 id="from-the-vcf-file-ogob-all-msu7-chr6gatkvariantfiltration-100000vcf">From the vcf file OgOb-all-MSU7-CHR6.GATKVARIANTFILTRATION-100000.vcf</h4>
<ul>
  <li>Transform the vcf file in a coordinate file <code class="highlighter-rouge">chr\tpos\tpos</code></li>
  <li>Now, in the VCF file, we would like to replace the genotypes by allelic dose. This means that we should replace the whole field by <code class="highlighter-rouge">0</code> when the genotype is <code class="highlighter-rouge">0/0</code>, by <code class="highlighter-rouge">1</code> when the genotype is <code class="highlighter-rouge">0/1</code> and <code class="highlighter-rouge">2</code> when the genotype is <code class="highlighter-rouge">1/1</code></li>
</ul>

<h4 id="from-fasta-files-in-linux-tpfasta">From fasta files in <code class="highlighter-rouge">LINUX-TP/Fasta</code></h4>
<ul>
  <li>In <code class="highlighter-rouge">fasta</code> directory, there are two files : <code class="highlighter-rouge">C_AllContigs.fasta</code> and <code class="highlighter-rouge">contig_tgicl.fasta</code>. Before to generate a unique file with all 2 libraries, we would like to tag each sequence per its origin. In each file, add the respective tag VS1- / VS2- just before the identifier.</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c"># File C_AllContigs.fasta initially</span>
<span class="o">&gt;</span>C_pseu_c1
AAAAATGTTTGAAATCCACTTGGCATTMAATGGTGAAAGAATTTTAGATTTTTATATACT
CCCTCGGTAAGGAAATTGTTGTCTCATTTTGGGATTCACAATTATTACCAACAGTGCAAG
GGTTTT

<span class="c">#File C_AllContigs.fasta</span>
<span class="o">&gt;</span>VS1-C_pseu_c1
AAAAATGTTTGAAATCCACTTGGCATTMAATGGTGAAAGAATTTTAGATTTTTATATACT
CCCTCGGTAAGGAAATTGTTGTCTCATTTTGGGATTCACAATTATTACCAACAGTGCAAG
GGTTTT</code></pre></figure>

<p>Rq : Test first the sed command on one file and STDOUT, then store the results in new files named RN-VS.MID1.clean.sff.fasta …</p>

<ul>
  <li>Generate a file named all-contigs.fasta with all the sequences - <code class="highlighter-rouge">cat file1 file2 &gt; file3</code></li>
  <li>Count the number of sequences in the fasta file just created <code class="highlighter-rouge">grep -c "&gt;" </code></li>
  <li>Count the sequence number of each library in this file</li>
</ul>

<h4 id="from-fastq-files-in-datafastqpairedtwoindividusirigin">From fastq files in <code class="highlighter-rouge">Data/fastq/pairedTwoIndividusIrigin</code></h4>
<ul>
  <li>In the directory <code class="highlighter-rouge">Data/fastq/pairedTwoIndividusIrigin</code> transform the fastq file irigin1_1.fastq in fasta format</li>
  <li>In one command line transform all fastq files of the directory in fasta (save the files before)</li>
</ul>

<hr />

<p><a name="practice-10"></a></p>
<h3 id="practice-10--manipulating-files-with-awk">Practice 10 : Manipulating files with <code class="highlighter-rouge">awk</code></h3>

<h4 id="from-the-gff-file-precedently-downloaded-1">From the gff file precedently downloaded</h4>
<ul>
  <li>Extract the coordinate from the gff file</li>
  <li>Calculate the mean of the gene length</li>
  <li>Calculate the mean of the gene length for the chromosome 1</li>
  <li>Count the number of genes above 2000bp length</li>
  <li>Bonus: calculate the mean of gene length for each chromosomes in one command line</li>
</ul>

<h4 id="from-the-result-of-a-nucmer-analysis">From the result of a nucmer analysis</h4>

<p>We want to rapidly align an assembly against a entire genome using nucmer. (i.e., assembling etc.) to a reference genome. Type the three following commands :</p>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="c">#So we compare one multifasta that have been created against a genome</span>
nucmer <span class="nt">--mum</span> reference.fasta contigs.fasta <span class="nt">-p</span> ctgVSref.NUCMER


<span class="c">#The previous command produces a file named ctgVSref.NUCMER.delta that can then be filtered using delta-filter and formatted using show-coords to produce a human-readable table of overlapping alignments between the two multifastas.</span>

<span class="c">#Filtering the nucmer results </span>
<span class="c">#The -l in delta-filter sets the minimum alignment length to 300. The -q “Maps each position of each query to its best hit in the reference, allowing for reference overlaps”.</span>
delta-filter <span class="nt">-l300</span> <span class="nt">-q</span> ctgVSref.NUCMER.delta <span class="o">&gt;</span> ctgVSref.filter300.delta

<span class="c">#Generate results (tab format)</span>
<span class="c">#The -c and -l in show-coords indicate that percent identity and sequence length information, respectively, should be included in the output. -L sets the minimum alignment length to display, -r sorts the output lines by reference IDs and coordinates, and -T switches the output to tab-delimited format.</span>
show-coords <span class="nt">-c</span> <span class="nt">-l</span> <span class="nt">-L</span> 300 <span class="nt">-r</span> <span class="nt">-T</span> ctgOMAP.filter300.delta <span class="o">&gt;</span> ctgOMAP.filter300.delta.coords.txt</code></pre></figure>

<ul>
  <li>Count the number of contigs in the fasta file</li>
  <li>Count the number of alignements performed by nucmer</li>
  <li>Count the number of contigs that have been aligned</li>
  <li>sort by alignment percent ascending</li>
  <li>count the number of alignement with alignment % &gt; 50 then 80</li>
</ul>

<hr />

<p><a name="practice-11"></a></p>
<h3 id="practice-11">Practice 11</h3>

<ul>
  <li>Go into the directory <code class="highlighter-rouge">LINUX-TP/Data/fastq/pairedTwoIndividusGzippedIrigin</code> - <code class="highlighter-rouge">cd</code></li>
  <li>List the directory content</li>
  <li>Run fastq-stats program ( <a href="http://manpages.ubuntu.com/manpages/xenial/man1/fastq-stats.1.html">more</a> to get stats about the fastq file <code class="highlighter-rouge">irigin1_1.fastq.gz</code></li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash">fastq-stats <span class="nt">-D</span> irigin1_1.fastq.gz</code></pre></figure>

<ul>
  <li>Use a <code class="highlighter-rouge">for</code> loop to run fastq-stats with every fastq file in the directory</li>
</ul>

<figure class="highlight"><pre><code class="language-bash" data-lang="bash"><span class="k">for </span>file <span class="k">in</span> <span class="k">*</span>fastq<span class="p">;</span> <span class="k">do 
  </span>fastq-stats <span class="nt">-D</span> <span class="nv">$file</span> <span class="o">&gt;</span> <span class="nv">$file</span>.fastq-stats <span class="p">;</span> 
<span class="k">done</span><span class="p">;</span></code></pre></figure>

<hr />

<h3 id="links">Links</h3>
<p><a name="links"></a></p>

<ul>
  <li>Related courses : <a href="https://southgreenplatform.github.io/trainings/linuxJedi/">Linux for Jedi</a></li>
  <li>Tutorials : <a href="https://southgreenplatform.github.io/trainings/linux/linuxTuto/">Linux Command-Line Cheat Sheet</a></li>
</ul>

<hr />

<h3 id="license">License</h3>
<p><a name="license"></a></p>

<div>
The resource material is licensed under the Creative Commons Attribution 4.0 International License (<a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">here</a>).
<center><img width="25%" class="img-responsive" src="http://creativecommons.org.nz/wp-content/uploads/2012/05/by-nc-sa1.png" />
</center>
</div>


	</div> <!-- #main -->
</div> <!-- #main-container -->
    <!-- Placed at the end of the document so the pages load faster -->
    <script src="http://localhost:4000/js/jquery-1.11.3.min.js"></script>
    <script src="http://localhost:4000/js/bootstrap.min.js"></script>
    <script src="http://localhost:4000/js/homepage.js"></script>
    <script src="http://localhost:4000/js/libs/modernizr-2.0.6.min.js"></script>

</script>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <div id="footer">
			<div class="logo-div" class="clearfix">
				<a class="logo" href="http://www.ird.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo_ird.png" alt="IRD Logo"  width="" />
				</a>

				<a class="logo" href="http://www.cirad.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo-cirad.png" alt="CIRAD Logo"   width="" />
				</a>

				<a class="logo" href="http://www.inra.fr">
					<img class="img-logo" src="http://localhost:4000/images/logo_inra.png" alt="INRA Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.bioversityinternational.org/">
					<img class="img-logo" src="http://localhost:4000/images/logo_bioversity.png" alt="Bioversity Logo"   width=""/>
				</a>

				<a class="logo" href="https://www.supagro.fr/">
					<img class="img-logo" src="http://localhost:4000/images/logo_supagro.png" alt="SupAgro Logo"   width=""/>
				</a>
				
				<a class="logo" href="http://www.southgreen.fr">
					<img class="img-logo" src="http://localhost:4000/images/southgreenlong.png" alt="South Green Logo"   width=""/>
				</a>

			</div>
		</div>

		<div class="footer_txt">
			<p>The contents of this website are &copy;&nbsp;2019 under the terms of the Licencied under
				<a href="http://www.cecill.info/licences/Licence_CeCILL-C_V1-en.html">CeCill-C</a>
				 and <a href="/blob/master/LICENSE">GPLv3</a>.
			</p>
		</div>
      </footer>
    </div>

</body>
</html>
